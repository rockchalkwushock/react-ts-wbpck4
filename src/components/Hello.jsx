import React from 'react'

const Hello = ({ framework, compiler }) => (
  <p>
    Welcome to {framework} with {compiler}
  </p>
)

export default Hello
