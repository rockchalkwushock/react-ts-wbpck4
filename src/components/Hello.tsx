import * as React from 'react'
import styled from 'styled-components'

export interface HelloProps {
  compiler: string
  framework: string
  transpiler: string
  types: string
}

const P = styled.p`
  font-size: 2rem;
  color: magenta;
  text-decoration: underline;
`

export class Hello extends React.Component<HelloProps, {}> {
  render() {
    const { compiler, framework, transpiler, types } = this.props
    return (
      <P>
        Welcome to {framework} with {compiler}, {transpiler}, and {types}.
      </P>
    )
  }
}
