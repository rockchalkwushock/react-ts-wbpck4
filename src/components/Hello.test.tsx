import * as React from 'react'
import { render } from 'react-testing-library'

import { Hello } from './Hello'

const Component = (
  <Hello
    compiler="Webpack 4"
    framework="React"
    transpiler="Babel 7"
    types="TypeScript"
  />
)

describe('<Hello />', () => {
  test('should render', () => {
    const { asFragment } = render(Component)
    expect(asFragment()).toMatchSnapshot()
  })
  test('should render correct text', () => {
    const { container } = render(Component)
    console.log(container.innerHTML)
    expect(container.innerHTML).toEqual(
      '<p>Welcome to React with Webpack 4, Babel 7, and TypeScript.</p>'
    )
  })
})
