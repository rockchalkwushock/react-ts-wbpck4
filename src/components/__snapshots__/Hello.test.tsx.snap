// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hello /> should render 1`] = `
<DocumentFragment>
  <p
    class="sc-bdVaJa hMjouN"
  >
    Welcome to React with Webpack 4, Babel 7, and TypeScript.
  </p>
</DocumentFragment>
`;
